### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by rock3r**
## Problem

A bunch of old dependencies need to be updated

## Solution

Update them! We're holding off on target/compile SDK update as we're going to migrate to androidx and bump the target/compile SDK in a separate PR, since it'll be noisy.

### Test(s) added

Nothing to test, but the app builds and runs.

### Paired with

Noobody

### CODE REVIEW

File: dependencies.gradle [hunk 1]
```diff
@@ -1,71 +1,66 @@
...
+            firebaseFirestore : 'com.google.firebase:firebase-firestore:17.0.2',
+            firebaseMessaging : 'com.google.firebase:firebase-messaging:17.0.0',
+            firebasePerf      : 'com.google.firebase:firebase-perf:16.0.0',
+            firebaseStorage   : 'com.google.firebase:firebase-storage:16.0.1',
+            firebaseUiStorage : 'com.firebaseui:firebase-ui-storage:4.0.1',
+            flexbox           : 'com.google.android:flexbox:1.0.0',
+            glide             : "com.github.bumptech.glide:glide:${glideVersion}",
+            glideCompiler     : "com.github.bumptech.glide:compiler:${glideVersion}",
+            glideOkHttp3      : "com.github.bumptech.glide:okhttp3-integration:${glideVersion}",
+            jodaTimeAndroid   : 'net.danlew:android.joda:2.9.9.4',
```
Comment (fourlastor) [inline dependencies.gradle:37]: worth having an issue to use JSR-310/TreeTenABP?

File: team-props/static-analysis/detekt-config.yml [hunk 2]
```diff
@@ -74,14 +76,15 @@ complexity:
   active: true
   ComplexCondition:
     active: true
-    threshold: 3
+    threshold: 4
```
Comment (fourlastor) [inline team-props/static-analysis/detekt-config.yml:79]: where is this coming from?

**Review by fourlastor — APPROVED** [review]

File: dependencies.gradle (hunk 1 above)
Comment (rock3r) [reply dependencies.gradle:37]: Yeah we should migrate eventually. Can you file it please?

File: team-props/static-analysis/detekt-config.yml (hunk 2 above)
Comment (rock3r) [reply team-props/static-analysis/detekt-config.yml:79]: Just merging in the defaults, if the thresholds are higher than what we have I raised to match

**Pull request merged**